src/components/Footer/index.ts

```diff
@@ -0,0 +1,72 @@
+import * as t from 'io-ts';
+
+export type LinkType = 'internal' | 'external';
+
+export interface FooterLinksType {
+  label: string;
+  /** the url to witch the user will be redirect */
+  href?: string;
+  ariaLabel: string;
+  linkType: LinkType;
+  /** if defined it will override the href behavior */
+  onClick?: () => void;
+}
+
+export interface PreLoginFooterSingleSectionType {
+  title?: string;
+  links: FooterLinksType[];
+}
+
+export interface PreLoginFooterSocialLink {
+  icon: string;
+  /** the url to witch the user will be redirect */
+  href?: string;
+  title: string;
+  ariaLabel: string;
+  /** if defined it will override the href behavior */
+  onClick?: () => void;
+}
+
+export interface PreLoginFooterLinksType {
+  aboutUs: PreLoginFooterSingleSectionType;
+  resources: PreLoginFooterSingleSectionType;
+  followUs: {
+    title: string;
+    socialLinks: PreLoginFooterSocialLink[];
+    links: FooterLinksType[];
+  };
+}
+
+export interface CompanyLinkType {
+  /** the url to witch the user will be redirect */
+  href?: string;
+  ariaLabel: string;
+  /** if defined it will override the href behavior */
+  onClick?: () => void;
+}
+
```

Comment:
Can you add a comment explaining what this class does and why do you need it?

Reply:
in case you need to add a function